R/post-body.R

```diff
@@ -24,3 +34,33 @@ parseBody <- function(body){
   }
   ret
 }
+
+
+#' @importFrom utils URLdecode
+#' @noRd
+parseMultipart <- function(body, boundary){
+  function(val, req, res, errorHandler){
+    tryCatch({
+      if (!requireNamespace("webutils", quietly = TRUE)) {
+        stop("The webutils package is not available but is required in order
+          to use the functionality to POST binary files",
+             call. = FALSE)
+      }
+      # Is there data in the request?
+      if (is.null(body) || length(body) == 0 || body == "") {
+        return(list())
+      }
+      parsed_binary <- webutils::parse_multipart(body, boundary)
+      file_name <- parsed_binary$myfile$filename
+      file_data <- parsed_binary$myfile$value
+
+      tmpfile <- tempfile()
+      writeBin(file_data, tmpfile)
+
+      ret <- NULL
```

Comment:
Is this any different than `ret <-list(name = file_name, data=tmpfile)`?

It surprised me that you could assign to a field on a NULL objec.t